charts/controls/Controls.tsx

```diff
@@ -435,109 +443,81 @@ class TimelineControl extends React.Component<{ chart: ChartConfig }> {
+            )
+        if (activeTab === "map") return chart.mapTransform.targetYearProp
+        else return chart.activeTransform.startYear!
+    }
 
-            const startYear =
-                dataTableTransform.autoSelectedStartYear ?? chart.timeDomain[0]
-            const endYear = chart.multiMetricTableMode
-                ? startYear
+    @computed get endYear() {
+        const { activeTab, chart } = this.props
+        if (activeTab === "table")
+            return chart.multiMetricTableMode
+                ? chart.dataTableTransform.startYear
                 : chart.timeDomain[1]
+        if (activeTab === "map") return chart.mapTransform.targetYearProp
+        else return chart.activeTransform.endYear!
+    }
 
+    componentDidUpdate(prevProps: TimelineControlProps) {
+        if (
+            prevProps.activeTab !== this.props.activeTab &&
+            this.props.activeTab != "map"
+        )
+            this.onChartTargetChange({
+                targetStartYear: this.startYear,
+                targetEndYear: this.endYear
+            })
+    }
+
+    @computed get timelineProps(): TimelineProps {
+        const { activeTab, chart } = this.props
+        return {
+            chart: chart,
+            years:
+                activeTab === "map"
+                    ? chart.mapTransform.timelineYears
+                    : chart.activeTransform.timelineYears,
+            startYear: this.startYear,
+            endYear: this.endYear,
+            onTargetChange: this.onChartTargetChange,
+            onStartDrag: this.onTimelineStart,
+            onStopDrag: this.onTimelineStop
+        }
+    }
+
+    render() {
+        if (this.timelineProps.years.length === 0) return null
+
+        const { activeTab, chart } = this.props
+
+        if (activeTab === "map")
             return (
                 <Timeline
-                    chart={chart}
-                    years={years}
-                    onTargetChange={this.onChartTargetChange}
-                    startYear={startYear}
-                    endYear={endYear}
-                    onStartDrag={this.onTimelineStart}
-                    onStopDrag={this.onTimelineStop}
-                    singleYearMode={chart.multiMetricTableMode}
-                />
-            )
-        } else if (chart.props.tab === "map") {
-            const { mapTransform } = chart
-            const years = mapTransform.timelineYears
-            if (years.length === 0) {
-                return null
-            }
-            return (
-                <Timeline
-                    chart={chart}
-                    years={years}
+                    {...this.timelineProps}
                     onTargetChange={this.onMapTargetChange}
-                    startYear={mapTransform.targetYearProp}
-                    endYear={mapTransform.targetYearProp}
                     singleYearMode={true}
+                    onStartDrag={undefined}
+                    onStopDrag={undefined}
```

Comment:
Do we really need that?
Maps should behave with `useTimelineDomains` set, right?

Reply:
`useTimelineDomains` is used to achieve the "zoom-out" effect on drag that we see on the e.g. line chart, so we definitely don't need it for the map. I prefer to remove it, so that someone debugging map doesn't see it as a prop and need to investigate it.